src/finchlite/autoschedule/compiler.py

```diff
@@ -191,6 +192,36 @@ class LogicLowerer:
     def __init__(self, mode: str = "fast"):
         self.mode = mode
 
+    @overload
```

Comment:
Once we decide to use parametric types we can get rid of these `@overload`s.

Reply:
For now, I'll leave the overloads, but later we should probably use parametric types. I just didn't want to use them in the IR nodes.

Reply:
Makes sense!